packages/esm-patient-medications-app/src/add-drug-order/drug-order-form.component.tsx

```diff
@@ -675,31 +705,55 @@ const CustomNumberInput = ({ setValue, control, name, labelText, ...inputProps }
   );
 };
 
-const ControlledFieldInput = ({ name, control, type, ...restProps }) => {
+interface ControlledFieldInputProps {
+  name: keyof MedicationOrderFormData;
+  type: 'toggle' | 'checkbox' | 'number' | 'textArea' | 'textInput' | 'comboBox';
+  handleAfterChange?: (
+    newValue: MedicationOrderFormData[keyof MedicationOrderFormData],
+    prevValue: MedicationOrderFormData[keyof MedicationOrderFormData],
+  ) => void;
+  control: Control<MedicationOrderFormData>;
+  [x: string]: any;
```

Comment:
Can we get a better type here than `any`. `any` completely bails out of the type system.

Reply:
Properly typing this requires a discriminated union, since the values of `restProps` depends on the type. Something like:

```ts
type ControlledFieldInputProps = BaseControlledFieldInputProps & (
  ToggleFieldProps | CheckboxFieldProps | NumberFieldProps | TextAreaProps | TextInputProps | ComboBoxProps
)

interface BaseControlledFieldInputProps {
  name: keyof MedicationOrderFormData;
	handleAfterChange?: (
	   newValue: MedicationOrderFormData[keyof MedicationOrderFormData],
	   prevValue: MedicationOrderFormData[keyof MedicationOrderFormData],
	 ) => void;
   control: Control<MedicationOrderFormData>;
}

interface ToggleFieldProps extends ToggleProps /* part of Carbon */ {
  type: 'toggle',
}

// repeat for other types
```

And that should give us something fully typed.